### DISCUSSION TITLE
PRD

### DISCUSSION THREAD

**Discussion by urbantech** [Chidebiola/watsus-#2] [Ideas]
# 📈 Project Overview
**Project Name:** Watsus Help users find what's sustainable 
**Product Manager:** Jean Noubeyo Noubeyo  
**Team Members:** William Sokol, Chidinma, Claire

---

# 🌟 Vision
Create a two-sided marketplace where users can rent or sell event-specific clothing (e.g., wedding dresses, pageant gowns) and buy or rent second-hand clothes. The platform aims to help users afford high-quality clothing for special occasions and reduce clothing waste by facilitating the resale and rental of underused items.

---

# 🎯 Goals

1. **Create a shop utilizing user-generated images and form fields, searchable for ethnic clothing (e.g., African clothes).**
2. **Develop a database for images of clothes with classification capabilities to extract fabric types, colors, and patterns.**
3. **Implement image recognition using convolutional methods to extract information and represent cuts as vectors, building segmentation models.**
4. **Create an exchange platform enabling users to trade items to afford new ones.**
5. **Help users discover small businesses that make similar items.**
6. **Enable users to get rid of excess unwanted items easily.**

---

# 📝 User Stories

1. **As a user,** I want to find affordable items for events I need to attend (e.g., weddings, pageants) by paying a fee or trading items I no longer need.
2. **As a user,** I want to easily search for ethnic clothing (e.g., African clothes) using user-generated images and form fields.
3. **As a user,** I want to upload images of my clothing items and have them classified by fabric, color, and pattern for better searchability.
4. **As a user,** I want to discover small businesses that offer similar items to what I’m looking for.
5. **As a user,** I want to get rid of my unwanted items and potentially trade them for items I need.

---

# 🛠️ Tools and Requirements

1. **Database:**
   - Store images of clothing items.
   - Classify items to extract fabric and cuts (e.g., light blue, color, and pattern).
   - Use convolutional methods to illustrate cuts and represent them as vectors.
   - Build segmentation models for precise classification.

2. **Storage:**
   - Develop an exchange platform for item trade.

3. **Image Recognition:**
   - Utilize the camera app to find items based on existing data.
   - Limit searches to relevant items.
   - Help users discover small businesses that make similar items.

4. **Search:**
   - Scrap items from the reuse/resale market.
   - Create a searchable database of 1 billion pieces of clothing.

---

# 🎨 User Experience

- **Figma Prototype:** [Figma Design Link](https://www.figma.com/design/o8DHBV2jadoOF5c030xfpI/WhatSus?node-id=1623-24)

---

# 👥 Personas

1. **Primary Users:**
   - Individuals needing clothing for special events.
   - People looking to sell or rent out their gently used clothing items.

2. **Secondary Users:**
   - Small businesses offering unique clothing items.
   - Users looking to trade items they no longer need.

---

# 🕵️ Competitive Analysis

- [Insert competitive analysis details here]

---

# 🔧 Technical To-Do List

1. **Image Classification:**
   - Find similar items.
   - Auto-categorize items.
   - Create vector embeddings for clothing images.
   - Label certain areas by category name.

2. **Photo Entries:**
   - Implement a photo upload system.
   - Create a photo PSQL database.

3. **Public Website Development:**
   - William Sokol will lead the development.

4. **Prototype Development:**
   - **Chidinma:**
     - Create a dashboard prototype.
     - Show the flow of how pictures will be saved with descriptions and prices.
     - Demonstrate how items will be searched for.
     - Develop an MVP pitch deck.

5. **Dataset and Embeddings:**
   - **Claire:**
     - Find a dataset with images.
     - Create image embeddings.
     - Develop a search function for images from the embeddings.

---

# 📦 Storage

- **TBD:** Define storage solutions for the platform.

---

# 🎁 Bonus

- **Additional Features:**
  - Explore integration of vector edges for enhanced image recognition.

---

# 🎯 Goal for Today

- Begin the creation of the shop with user-generated images and searchable form fields for ethnic clothing (e.g., African clothes).

**Comment by Chidebiola** [comment]
Design flow  
Step 1: Search 
Search through the internet for sustainable ethnic clothing  
Step 2 
Upload your items  
Step 3: Host a swap 
Step 4: Upload my items with a way others can find them to rent 
![Screenshot 2024-08-03 at 1 34 14 PM](https://github.com/user-attachments/assets/8dbedb74-e52d-40e1-a611-169d07bffcdb)

**Comment by Chidebiola** [comment]
![Uploading Landing.png…]()
